developing/docker/README.md

```diff
@@ -94,6 +98,162 @@ To get a `top` like report of what your containers are doing
+    docker image inspect hello-world:latest
+
+The output will look something like this (not all fields are shown):
+
+    [
+        {
+            "Id": "sha256:bf756fb1ae65adf866bd8c456593cd24beb6a0a061dedf42b26a993176745f6b",
+            "RepoTags": [
+                "hello-world:latest"
+            ],
+            "RepoDigests": [
+                "hello-world@sha256:4cf9c47f86df71d48364001ede3a4fcd85ae80ce02ebad74156906caff5378bc"
+            ],
+            "Parent": "",
+            ...
+            "Container": "71237a2659e6419aee44fc0b51ffbd12859d1a50ba202e02c2586ed999def583",
+            "ContainerConfig": {
+                ...
+                "Image": "sha256:eb850c6a1aedb3d5c62c3a484ff01b6b4aade130b950e3bf3e9c016f17f70c34",
+                ...
+            },
+            ...
+            "Config": {
+                ...
+                "Image": "sha256:eb850c6a1aedb3d5c62c3a484ff01b6b4aade130b950e3bf3e9c016f17f70c34",
+                ...
+            },
+            ...
+            "RootFS": {
+                "Type": "layers",
+                "Layers": [
+                    "sha256:9c27e219663c25e0f28493790cc0b88bc973ba3b1686355f221c38a36978ac63"
+                ]
+            },
+        }
+    ]
+
+In this example, there are four unique sha256 digests: Id, RepoDigests,
+ContainerConfig/Config (which share the same sha256), and Layers (many images
+will have more one than layer, each with its own sha256).
+
+For comparison, here is the `docker image ls` output for the same image:
+
+    docker image ls --digests hello-world
+
+Output:
+
+    REPOSITORY          TAG                 DIGEST                                                                    IMAGE ID            CREATED             SIZE
+    hello-world         latest              sha256:4cf9c47f86df71d48364001ede3a4fcd85ae80ce02ebad74156906caff5378bc   bf756fb1ae65        9 months ago        13.3kB
+
+The following table shows how the digests and IDs correspond between the output
+of the `docker image ls` and `docker image inspect` commands.
+
+| docker image ls| docker image inspect |
+| :------------- | :------------------- |
+| DIGEST | RepoDigests |
+| IMAGE ID | Id |
+
+Why are there so many sha256 digests? Why are they different? Which one(s) will
+you most likely use in your daily workflow?
+
+Let's start by reviewing some of the terminology involved:
+
+    Image: a collection of one more layers.
+
+    Layer: identified by a digest, which is a computed SHA256 of the layer's
+    content. If the content changes, the digest will change.
+
+    Image configuration object (JSON): includes, among other things, an ordered
+    list of all the layer digests that combine to form an image.
+
+Now, let's go through some of the fields in the `docker image inspect` output:
+
+    Id: this is the Image ID, a digest or computed SHA256 hash of the image's JSON
+    configuration object. Note: this is different from the RepoDigest.
```

Comment:
The JSON configuration object contains information about the local machine and thus that is why the `Id` is not globally unique.